.vsts-pr.yml

```diff
@@ -0,0 +1,150 @@
+          - name: _SignType
+            value: real
+          - name: _OfficialBuildArgs
+            value: /p:DotNetPublishUsingPipelines=true
+                   /p:DotNetSignType=$(_SignType)
+                   /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+                   /p:TeamName=$(_TeamName)
+        - ${{ else }}:
+          - name: _SignType
+            value: test
+          - name: _OfficialBuildArgs
+            value: ''
+        strategy:
+          matrix:
+            Release:
+              _BuildConfig: Release
+        steps:
+        - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+          - task: CodeQL3000Init@0
+            displayName: Initialize CodeQL
+            condition: and(succeeded(), eq(variables['Codeql.Enabled'], 'true'))
+        - script: eng\common\CIBuild.cmd
+            -configuration $(_BuildConfig)
+            -prepareMachine
+            $(_OfficialBuildArgs)
+          name: Build
+          displayName: Build and run tests
+          condition: succeeded()
+        - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+          - task: CodeQL3000Finalize@0
+            displayName: Finalize CodeQL
+            condition: and(succeeded(), eq(variables['Codeql.Enabled'], 'true'))
+          # Guardian requires npm.
+          - task: NodeTool@0
+            inputs:
+              versionSpec: '18.x'
+          # Validates compiler/linker settings and other security-related binary characteristics.
+          # https://github.com/Microsoft/binskim
+          # YAML reference: https://eng.ms/docs/security-compliance-identity-and-management-scim/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/sdl-azdo-extension/binskim-build-task#v4
+          - task: BinSkim@4
+            displayName: Run BinSkim
+            inputs:
+              InputType: Basic
+              Function: analyze
+              TargetPattern: binskimPattern
+              AnalyzeTargetBinskim: $(Build.SourcesDirectory)\artifacts\bin\Sign.Cli\$(_BuildConfig)\net8.0\win-x64\publish\*.dll
+              AnalyzeSymPath: 'SRV*https://symweb'
+            condition: succeededOrFailed()
+          - task: PublishTestResults@2
+            displayName: 'Publish Unit Test Results'
+            inputs:
+              testResultsFormat: xUnit
+              testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
+              mergeTestResults: true
+              searchFolder: $(System.DefaultWorkingDirectory)
+              testRunTitle: sign unit tests - $(Agent.JobName)
+            condition: succeededOrFailed()
+          - task: ComponentGovernanceComponentDetection@0
+            displayName: Component Governance scan
+            inputs:
+              ignoreDirectories: '$(Build.SourcesDirectory)/.packages,$(Build.SourcesDirectory)/artifacts/obj/Sign.Cli'
+
+- ${{ if eq(variables['System.TeamProject'], 'internal') }}:
```

Comment:
Will this condition ever be true? I wonder if these make the pipeline more complex for no reason

Reply:
Yes, in the internal pipeline.

After this merges, I will create a new PR to simplify this file by removing official-only stuff.